xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-dialog/modal.js

```diff
@@ -58,8 +59,20 @@ define('modal', ['jquery', 'l10n!modal', 'bootstrap'], function($, translations)
     });
     modal.find('.modal-title').text(definition.title);
     modal.find('.modal-body').html('').append(definition.content);
-    modal.find('.modal-footer .btn-primary').text(definition.acceptLabel);
+    const submitButton = modal.find('.modal-footer .btn-primary');
+    submitButton.text(definition.acceptLabel);
     modal.find('.modal-footer .btn[data-dismiss="modal"]').text(definition.dismissLabel);
+    if(definition.form) {
+      // We want the submit button to actually submit the form
+      submitButton.removeAttr('type');
+      // This modal should act and look like a form. We replace its content node by a form, so we can benefit from
+      // native form utilities, such as implicit form validation (pressing Enter on a text field)
+      const modalContent= modal.find('.modal-content');
+      const modalContentInnerHTML = modalContent.html();
+      const formModalContent = $('<form class="modal-content">');
+      formModalContent.html(modalContentInnerHTML);
+      modalContent.replaceWith(formModalContent);
```

Comment:
```suggestion
      const formModalContent = $('<form class="modal-content">');
      formModalContent.append(modalContent.contents());
      modalContent.replaceWith(formModalContent);
```

It's safer and faster to move the existing nodes rather than using ``innerHTML``.